### PR TITLE
Fix prefer-destructuring issues

### PR DESCRIPTION
**Pull request by whymarrh**
Refs #8982

See [`prefer-destructuring`](https://eslint.org/docs/rules/prefer-destructuring) for more information.

This change enables `prefer-destructuring` and fixes the issues raised by the rule.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -70,6 +70,18 @@ module.exports = {
...
+      'VariableDeclarator': {
+        'array': false,
+        'object': true,
+      },
+      'AssignmentExpression': {
+        'array': false,
+        'object': false,
+      },
+    }, {
+      'enforceForRenamedProperties': false,
```
Comment (rekmarks) [inline .eslintrc.js:83]: Why not `true`? 🤔

Comment (whymarrh) [reply .eslintrc.js:83]: Enabling it seems to conflict with turning off `VariableDeclarator.array` and `AssignmentExpression.array` which in turn raises annoying issues, e.g.:

```js
const contract = contractMap[address]
```

Comment (whymarrh) [reply .eslintrc.js:83]: Huh, writing that out makes me realize that that isn't a conflict, that isn't array access. Fair enough.

Enabling it raises 1k+ issues that I'm too lazy to fix by hand.

Comment (whymarrh) [reply .eslintrc.js:83]: It is more verbose, though:

```diff
-const contract = contractMap[address]
+const { [address]: contract } = contractMap
```

**Review by rekmarks — APPROVED** [review]
LGTM!

**Comment by whymarrh** [comment]
I've created MetaMask/eslint-config#57 to update the base config

**Pull request merged**